module-system/sov-modules-api/src/default_signature.rs

```diff
@@ -3,70 +3,87 @@ use std::hash::Hash;
-use ed25519_dalek::ed25519::signature::Signature as DalekSignatureTrait;
 use ed25519_dalek::{
-    PublicKey as DalekPublicKey, Signature as DalekSignature, PUBLIC_KEY_LENGTH, SIGNATURE_LENGTH,
+    Signature as DalekSignature, VerifyingKey as DalekPublicKey, PUBLIC_KEY_LENGTH,
 };
-use serde::de::Error;
-use serde::{Deserialize, Deserializer, Serialize, Serializer};
 
 use crate::{SigVerificationError, Signature};
 
 #[cfg(feature = "native")]
 pub mod private_key {
-
-    use ed25519_dalek::{Keypair, SignatureError, Signer};
+    use ed25519_dalek::{Signer, SigningKey, KEYPAIR_LENGTH, SECRET_KEY_LENGTH};
     use rand::rngs::OsRng;
     use thiserror::Error;
 
     use super::{DefaultPublicKey, DefaultSignature};
     use crate::{Address, PrivateKey, PublicKey};
 
     #[derive(Error, Debug)]
-    pub enum DefaultPrivateKeyHexDeserializationError {
+    pub enum DefaultPrivateKeyDeserializationError {
         #[error("Hex deserialization error")]
         FromHexError(#[from] hex::FromHexError),
-        #[error("PrivateKey deserialization error")]
-        PrivateKeyError(#[from] SignatureError),
+        #[error("KeyPairError deserialization error")]
+        KeyPairError(#[from] ed25519_dalek::SignatureError),
+        #[error("Invalid private key length: {actual}, expected {expected_1} or {expected_2}")]
+        InvalidPrivateKeyLength {
+            expected_1: usize,
+            expected_2: usize,
+            actual: usize,
+        },
     }
 
     /// A private key for the default signature scheme.
     /// This struct also stores the corresponding public key.
+    #[derive(Clone, serde::Serialize, serde::Deserialize)]
     pub struct DefaultPrivateKey {
-        key_pair: Keypair,
+        key_pair: SigningKey,
     }
 
-    impl core::fmt::Debug for DefaultPrivateKey {
-        fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-            f.debug_struct("DefaultPrivateKey")
-                .field("public_key", &self.key_pair.public)
-                .field("private_key", &"***REDACTED***")
-                .finish()
+    impl DefaultPrivateKey {
+        fn try_from_keypair(value: &[u8]) -> Result<Self, DefaultPrivateKeyDeserializationError> {
+            let value: [u8; KEYPAIR_LENGTH] = value
+                .try_into()
+                .expect("incorrect usage of `try_from_keypair`, check input length");
+            let key_pair = SigningKey::from_keypair_bytes(&value)?;
+            Ok(Self { key_pair })
         }
-    }
 
-    impl serde::Serialize for DefaultPrivateKey {
-        fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-            serializer.serialize_bytes(&self.key_pair.to_bytes())
+        fn try_from_private_key(value: &[u8]) -> Self {
+            let value: [u8; SECRET_KEY_LENGTH] = value
+                .try_into()
+                .expect("incorrect usage of `try_from_private_key`, check input length");
```

Comment:
I'm not sure its safe to panic here as user input might lead to inconsistent states. If its safe, can we add a comment on why is that?

Reply:
Idea behind this panic is that it is private method only used by `try_from`, which currently checks length of user input. So if this panics, means that there's a bug in our code. 

But comment explaining this is definitely must have